pkg/virtctl/vnc/vnc.go

```diff
@@ -146,32 +159,74 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 		listenResChan <- err
 	}()
 
-	// execute remote viewer
+	// execute VNC
 	go func() {
+		defer close(doneChan)
 		port := ln.Addr().(*net.TCPAddr).Port
-		args := []string{fmt.Sprintf("vnc://127.0.0.1:%d", port)}
-		if glog.V(4) {
-			args = append(args, "--debug")
-			glog.Infof("remote-viewer commandline: %v", args)
-		}
+		args := []string{}
 
-		cmnd := exec.Command("remote-viewer", args...)
+		vncBin := ""
+		osType := runtime.GOOS
+		switch osType {
+		case "darwin":
+			if _, err := os.Stat(TIGER_VNC); err == nil {
+				vncBin = TIGER_VNC
+				args = tigerVncArgs(port)
+			} else if !os.IsNotExist(err) {
+				viewResChan <- err
+				return
+			} else if _, err := os.Stat(CHICKEN_VNC); err == nil {
+				vncBin = CHICKEN_VNC
+				args = chickenVncArgs(port)
+			} else if !os.IsNotExist(err) {
+				viewResChan <- err
+				return
+			} else if _, err := os.Stat(REAL_VNC); err == nil {
+				vncBin = REAL_VNC
+				args = realVncArgs(port)
+			} else if !os.IsNotExist(err) {
+				viewResChan <- err
+				return
+			}
```

Comment:
we should also handle case when remote-viewer was installed in osx and can be found in path

Reply:
The thing about the RemoteViewer package is that it seems to require a supplemental script on the user's side to get it to launch cleanly via CLI, I can't just 'ln -s /Applications/RemoteViewer.app/Contents/MacOS/RemoteViewer /usr/local/bin/remote-viewer' for example.

I do like the idea though of having a fallback of just looking for a 'remote-viewer' in the path as a fallback, people are free to script that to what they desire. I'll add the suggestion.

Reply:
I've pushed the suggested changes and re-tested with all four 'if' cases on MacOS now.  For the remote-viewer case I still get the error 'Unknown graphic type for the guest vnc://127.0.0.1:59476' as described in #812 but I'm just going to assume that is an issue with my quick attempt to get a custom remote-viewer working, and that if someone has a working remote-viewer setup in their path that it will work.